reference/conanfile/methods/finalize.rst

```diff
@@ -0,0 +1,74 @@
+.. _reference_conanfile_methods_finalize:
+
+finalize()
+==========
+
+.. include:: ../../../common/experimental_warning.inc
+
+Package immutability is an important concept in Conan. It ensures that the package is not modified after it has been built and packaged,
+so that the package id is consistent and the package can be reused in different machines.
+
+This method is intended for customization of the package in the running machine, allowing modifications to the package
+that will be used by the consumers of the package, but not modifying the original package inside the Conan cache.
+This method is called after the package has been installed in the local cache and
+the modifications are not uploaded to any remote server, they are only local.
+
+
+The main use-cases of this method include:
+
+- Having different packages that get modified when they are run (Like python tools that generate pycache files as part
+  of their execution)
+- Modifying the package to be used in the local machine (Like creating a ``conf`` file with the necessary
+  environment variables for the package to be used in the local machine)
+
+These changes are transparent to the consumers of the package, they will use the customized package as if they were dealing with the original package,
+so changes made in this method should work.
+
+
+finalize() example usage
+------------------------
+
+The most common use-case of this method is to avoid corrupting the immutability of the package.
+
+For example, if a package generates some files during its execution, like a python package that generates pycache files,
+you can use this method to generate those files in the local machine, without affecting the original package.
+This is the case for tools like Meson.
+
+
+.. code-block:: python
```

Comment:
This is a bit barren, I'd like to have a more complete example but don't know how much further to take it